BootloaderCorePkg/Stage1A/Stage1A.c

```diff
@@ -172,7 +173,13 @@ PrepareStage1B (
     if (IS_COMPRESSED (Src)) {
       Length = sizeof (LOADER_COMPRESSED_HEADER) + Hdr->CompressedSize;
     }
-    Status = DoHashVerify ((CONST UINT8 *)Src, Length, HASH_USAGE_STAGE_1B, HASH_TYPE_SHA256, NULL);
+    if(FixedPcdGet8(PcdCompSignHashAlg) == HASH_TYPE_SHA256){
+      SignHashAlg = HASH_TYPE_SHA256;
+    } else if(FixedPcdGet8(PcdCompSignHashAlg) == HASH_TYPE_SHA384){
+      SignHashAlg = HASH_TYPE_SHA384;
+    }
+
+    Status = DoHashVerify ((CONST UINT8 *)Src, Length, HASH_USAGE_STAGE_1B, SignHashAlg, NULL);

```

Comment:
can you use PcdGet8() directly in this call ?

Reply:
will do.